libpromises/evalfunction.c

```diff
@@ -238,8 +238,8 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             break;
 
         case RVAL_TYPE_CONTAINER:
-            convert = value;
-            *allocated = false;
+            convert = JsonCopy(value);
```

Comment:
But this change basically reverts small part of commit da9a9fc8ff4c0d65520b07c03077136a4947c4a2 that you have committed @tzz. I am not sure why this had to be changed with that commit, still reading it.